doc/topics/core-structure.md

```diff
@@ -0,0 +1,202 @@
+- `<subsystem>_wrap` files (Libuv wrappers for Proxying to/from JS APIs)
+- Other bootstrapping / utility `.cc`/`.h` files, particularly `node.cc` and `env-inl.h`.
+
+The following traces Node.js through startup, showing how it interacts with
+C++ & JS:
+
+Step | Description | Source Location
+---|---|---
+1 | startup: `main --> node::Start --> node::StartNodeInstance` | ./src/node_main.cc::51, ./src/node.cc::4345, ./src/node.cc::4383
+2 | create JSVM (`v8::isolate`) | ./src/node.cc::4372, ./src/node.cc::4261
+3 | create event loop (`uv_loop_t`) | ./src/node.cc::4377
+4 | create native environment (`node::Environment`) containing JSVM and event loop | ./src/node.cc::4276, ./src/node.cc::4181
+5 | create JS `process` object bound to native environment | ./src/node.cc::4237-4244
+6 | initialize JS context and `process` object through JS script | ./src/node.cc::4289, ./src/node.cc::3330, ./lib/internal/bootstrap_node.js
+7 | invoke user script (or repl, --eval, etc.) | ./lib/internal/bootstrap_node.js::66
+
+
+### Tests & Benchmarks (`test/` & `benchmark/`)
+
+Node.js's extensive test suite can be found within `test/`. Most new & existing
+tests can be found within `test/parallel/`, which is the directory we tell the
+test runner to parallelize over so that the test suite can run faster.
+
+Most times if you modify code within node, tests for that behavior will also be
+needed for the change to be accepted.
+
+The tests can be run via `make -j4 test` on Unix, or `vcbuild test` on Windows.
+Do note that Node.js will have to be built before. Please see [BUILDING][] for
+details on the build prerequisites.
+
+Additionally, Node.js also contains a benchmarks suite for performance-critical
+areas. Typically, new contributions do not require new benchmarks unless they
+are performance-specific changes. At the time of writing, the benchmarks suite
+is far from comprehensive and is in need of improvement.
+
+
+### Build / Internal Tooling (`tools/`)
+
+These are mostly contained within the `tools/` folder, but some tools files,
+mentioned previously, are top-level so they can be easily run from the root
+directory. If you are building node you will become quickly familiar with some
+of these.
+
+Other files here include the various gyp and native JavaScript embedding tools.
+Some of this is quite complex and specific in order to build V8 correctly, and
+also let node run on a vast array of different platforms. Lots of these are
+quite arcane in nature.
+
+The [gyp][] and [eslint][] tooling dependencies also live in the `tools/`
+folder. Patches to these must be sent to their upstream projects like the rest
+of our dependencies, as outlined in the section below.
+
+
+### Dependencies (`deps/`)
+
+Node.js is built on a number of dependencies that allow it to function. Most of
+these are "native" (C/C++), and must be compiled.
+
+Node Core does not directly take patches or pull requests to most of its
```

Comment:
Then I would say that explicitly rather than saying "We don't do this" when we do.